homework-submission/reactweek1/src/App.js

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react';
+import './App.css';
+
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.changeToDone = this.changeToDone.bind(this);
+    this.descriptionHandler = this.descriptionHandler.bind(this);
+    this.deadLineHandler = this.deadLineHandler.bind(this);
+    this.removeHandler = this.removeHandler.bind(this);
+    this.addHandler = this.addHandler.bind(this);
+    this.state = {
+      list: [
+        {
+          id: 1,
+          description: "Get out of bed",
+          deadline: "2017-09-11",
+          done: true
+        },
+        {
+          id: 2,
+          description: "Brush teeth",
+          deadline: "2017-09-10",
+          done: false
+        },
+        {
+          id: 3,
+          description: "Eat breakfast",
+          deadline: "2017-09-09",
+          done: false
+        }
+      ],
+      newDescription: "",
+      newDeadline: ""
+    };
+  }
+
+  changeToDone = event => {
+    const id = event.target.id;
+    const listCopy = [...this.state.list];
+    const itemIndex = listCopy.findIndex(el => {
+      return el.id === Number(id);
+    });
+    listCopy[itemIndex].done = !listCopy[itemIndex].done;
+    this.setState({
+      list: listCopy
+    });
+  };
+  descriptionHandler = newDescription => {
+    const listCopy = [...this.state];
+    listCopy["newDescription"] = newDescription;
+    this.setState(listCopy);
+  };
+
+  // descriptionHandler = newDescription => this.setState({ newDescription }); does not work properly
+
+  deadLineHandler = newDeadLine => {
+    const listCopy = [...this.state];
+    listCopy["newDeadline"] = newDeadLine;
+    this.setState(listCopy);
+  };
+
+  // deadLineHandler = newDeadLine => this.setState({ newDeadLine }); does not work properly
+
+  addHandler = () => {
+    const state = this.state;
+    const newItem = {
+      id: this.state.list.length + 1,
+      description: this.state.newDescription,
+      deadline: this.state.newDeadline,
+      done: false
+    };
+    ([...this.state.newDeadline] = ""), ([...this.state.newDescription] = "");
+
+    this.setState({ list: [...state.list, newItem] });
+  };
+
+  removeHandler = i => {
+    let state = this.state;
+    const updatedList = state.list.filter(prop => prop.id !== i + 1);
```

Comment:
why `i + 1`? this seems totally random :)
why don't you pass the id to this function instead of passing an index?

Reply:
I do not know, i just came up with that idea but your idea is indeed better.